projects/packages/my-jetpack/_inc/components/product-cards-section/boost-card/use-boost-tooltip-copy.ts

```diff
@@ -0,0 +1,103 @@
+import { __ } from '@wordpress/i18n';
+import React, { useMemo } from 'react';
+import { useProduct } from '../../../hooks/use-product';
+import usePurchases from '../../../hooks/use-purchases';
+
+const JETPACK_BOOST_PRODUCTS = [
+	'jetpack_boost_bi_yearly',
+	'jetpack_boost_yearly',
+	'jetpack_boost_monthly',
+];
+
+/**
+ * Gets the tooltip copy based on the Boost letter grade and other factors.
+ *
+ * @param {object} props - Component props.
+ * @param {string} props.speedLetterGrade - The Boost speed letter grade.
+ * @returns {React.ReactElement | string} A translated JSX Element or string.
+ */
+export function useBoostTooltipCopy( { speedLetterGrade } ) {
+	const slug = 'boost';
+	const { purchases, isFetchingPurchases } = usePurchases();
+	const hasBoostPaidPlan = useMemo( () => {
+		if ( isFetchingPurchases ) {
+			return null;
+		}
+		if ( ! purchases?.length ) {
+			return false;
+		}
+
+		return (
+			purchases.filter( purchase => JETPACK_BOOST_PRODUCTS.includes( purchase.product_slug ) )
+				.length > 0
+		);
+	}, [ isFetchingPurchases, purchases ] );
+
+	const { detail } = useProduct( slug );
+	const { isPluginActive } = detail;
+
+	// Boost plugin is active
+	if ( isPluginActive ) {
+		//  Has a paid Boost plan
+		if ( hasBoostPaidPlan ) {
+			switch ( speedLetterGrade ) {
+				case 'A':
+					return __(
+						'Your site is fast! Boost is working to enhance your performance with automated tools.',
+						'jetpack-my-jetpack'
+					);
+				case 'B':
+				case 'C':
+				case 'D':
+				case 'E':
+				case 'F':
+					return __(
+						'Visit the Boost dashboard to view your historical speed scores and manage your product settings.',
+						'jetpack-my-jetpack'
+					);
+				// This case is here to prevent build optimization minification breaking the build.
```

Comment:
What if we just had 

```jsx
case 'A':
					return __(
						'Your site is fast! Boost is working to enhance your performance with automated tools.',
						'jetpack-my-jetpack'
					);
default:
					return __(
						'Visit the Boost dashboard to view your site’s speed scores and manage your product settings.',
						'jetpack-my-jetpack'
					);
```

Was that what was breaking the build?

Reply:
Yes, that is what was breaking the build.  The build optimization process converts it to a ternary when we do it that way:, i.e.-
```
case === 'A'
   ? __( ''Your site is fast! Boost is working...', jetpack-my-jetpack )
   : __( 'Visit the Boost dashboard to view...', jetpack-my-jetpack );
```

Reply:
Good to know, thank you for clarifying!